presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -553,6 +562,15 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
             return Optional.of(mostSpecificFunction.getBoundSignature());
         }
 
+        // Prefer functions where all arguments are of the same type
```

Comment:
Yeah, but I think any behavior that makes a function with unknown arguments not ambiguous when it would have been ambiguous is going to be arbitrary and kind of a hack.  I'd say the same for creating separate unknown functions since the whole thing with unknown  is that it's not its own type so it doesn't have functions.  And there isn't an expected behavior because the function call is ambiguous. It seems though like most of the time you multiply or divide things by their own type.  And then at least you are calling a function with actual types.  So I think it's a reasonable choice.

When I looked at how postgresql handles ambiguous functions, it sometimes will return an ambiguous function error, but for cases  like this where it's choosing between arguments with the same type or different types, it would give you the function with arguments of the same type.